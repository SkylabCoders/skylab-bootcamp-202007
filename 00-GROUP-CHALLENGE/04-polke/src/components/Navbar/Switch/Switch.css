.switch {
	display: flex;
	margin-right: 8px;
	height: 28px;
}
.sun {
	margin-right: 9px;
}

@media only screen and (min-width: 600px) {
	.switch {
		height: 28px;
	}
}
