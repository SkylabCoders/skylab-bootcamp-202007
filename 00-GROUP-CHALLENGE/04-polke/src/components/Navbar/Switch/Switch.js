import React from 'react';
import './Switch.css';
import '../../../shared/generalStyles.css';

function Switch() {
	var isDarkActive = false;
	// codi obligatori x poder cridar a stylesheet
	var styleEl = document.createElement('style');
	document.head.appendChild(styleEl);

	var styleSheet = styleEl.sheet;
	// insertem la primera rule (light mode)

	styleSheet.insertRule(
		`:root {--bg-color-navbar: #24292e;
			--bg-color-box: #f8f9fa9f;
			--font-color-footer: #f8f9fa;
			--form-color-headers: #212529;
			--form-color-background: #bec0c5;
			--bg-color-body: #f5f5f5;
			--bg-color-button: #007bff;
			--bg-color-repoCard: #f8f9fa;
			--font-color-repoCard: black;
			--bg-color-repoCard__name-private: #f1f2f3;
			--bg-color-repoCard__details: #f1f2f3;
			--bg-color-user-info-card: #f8f9fa;
			--font-color-user-info-card: black;}`,
		0
	);
	// Functio que mira si esta actiu el darkmode o no, i canvia les regles en funcio d'aixo
	const switchColor = function () {
		if (isDarkActive) {
			isDarkActive = false;
			styleSheet.deleteRule(0);
			styleSheet.insertRule(
				`:root {
					--bg-color-navbar: #24292e;
					--bg-color-box: #f8f9fa9f;
					--bg-color-body: #f5f5f5;
					--bg-color-button: #007bff;
					--bg-color-repoCard: #f8f9fa;
					--bg-color-repoCard__name-private: #f1f2f3;
					--bg-color-repoCard__details: #f1f2f3;
					--bg-color-user-info-card: #f8f9fa;
					--font-color-footer: #f8f9fa;
					--font-color-repoCard: black;
					--font-color-user-info-card: black;
					--font-color-input:black;
					--font-color-carousel:black;
					--form-color-headers: #212529;
					--form-color-background: #bec0c5;
					;}`,
				0
			);
		} else {
			styleSheet.deleteRule(0);
			styleSheet.insertRule(
				`:root {
				--bg-color-repoCard: #4D5157;
				--bg-color-body: #20252D;
				--bg-color-navbar: #24292e;
				--bg-color-repoCard__name-private: #46494D;
			    --bg-color-repoCard__details: #46494D;
				--bg-color-user-info-card: #4D5157;	
				--bg-color-button: #007bff;
				--font-color-footer: #f8f9fa;
				--font-color-repoCard: white;
				--font-color-repoCard: white;
				--font-color-user-info-card: white;
				--font-color-input:white;
				--font-color-carousel:white;
				--form-color-headers: white;
				--form-color-background: #24292e;

				}`,
				0
			);
			isDarkActive = true;
		}
	};
	return (
		<div className="switch">
			<img
				src={require('../../../assets/img/sun.png')}
				alt="sun"
				className="sun"
			/>
			<div className="custom-control custom-switch">
				<input
					onChange={switchColor}
					type="checkbox"
					className="custom-control-input"
					id="customSwitches"
					readOnly
				/>
				<label
					className="custom-control-label	"
					htmlFor="customSwitches"
				></label>
			</div>
			<img
				src={require('../../../assets/img/moon.png')}
				alt="moon"
				className="moon"
			/>
		</div>
	);
}
export default Switch;
