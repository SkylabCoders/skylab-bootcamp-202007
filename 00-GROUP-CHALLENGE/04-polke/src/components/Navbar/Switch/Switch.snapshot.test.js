import renderer from 'react-test-renderer';
import Switch from './Switch';
import React from 'react';

describe('Switch snapshot', () => {
	const tree = renderer.create(<Switch />);
	it('should match', () => {
		expect(tree.toJSON()).toMatchSnapshot();
	});
});
