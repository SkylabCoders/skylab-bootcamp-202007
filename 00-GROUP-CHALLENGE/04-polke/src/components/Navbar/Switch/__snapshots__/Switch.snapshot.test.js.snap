// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch snapshot should match 1`] = `
<div
  className="switch"
>
  <img
    alt="sun"
    className="sun"
    src="sun.png"
  />
  <div
    className="custom-control custom-switch"
  >
    <input
      className="custom-control-input"
      id="customSwitches"
      onChange={[Function]}
      readOnly={true}
      type="checkbox"
    />
    <label
      className="custom-control-label	"
      htmlFor="customSwitches"
    />
  </div>
  <img
    alt="moon"
    className="moon"
    src="moon.png"
  />
</div>
`;
